src/lexactivator.ts

```
@@ -823,6 +833,46 @@ export class LexActivator {
 		return arrayToString(array);
 	}
 
+	/**
+	 * Checks whether a new release is available for the product.
+	 * 
+	 * This function should only be used if you manage your releases through
+	 * Cryptlex release management API.
+	 * 
+	 * When this function is called the release update callback function gets invoked 
+	 * which passes the following parameters:
+	 * 
+	 * status - determines if any update is available or not. It also determines whether 
+	 * an update is allowed or not. Expected values are LA_RELEASE_UPDATE_AVAILABLE,
+	 * LA_RELEASE_UPDATE_NOT_AVAILABLE, LA_RELEASE_UPDATE_AVAILABLE_NOT_ALLOWED.
+	 * 
+	 * releaseJson- returns json string of the latest available release, depending on the 
+	 * flag LA_RELEASES_ALLOWED or LA_RELEASES_ALL passed to the CheckReleaseUpdate().
+	 * 
+	 * @param {function(number, string)} releaseUpdateCallback callback function
+	 * @param {ReleaseFlags} flag If an update only related to the allowed release is required, 
+	 * then use LA_RELEASES_ALLOWED. Otherwise, if an update for all the releases is
+	 * required, then use LA_RELEASES_ALL.
+	 * @throws {LexActivatorException}
+	 */
+	static CheckReleaseUpdate(releaseUpdateCallback: ReleaseUpdateCallback, flag: typeof ReleaseFlags[keyof typeof ReleaseFlags]): void {
+		const internalReleaseUpdateCallback = function (status: number, releaseJson: string ): void {
+			let release: Release | {} = {};
+			if (releaseJson) {
+				try {
+					release = JSON.parse(releaseJson);
+				} catch (error:any) {
+					console.error(error.message);
```

a library never logs its own errors

---

Yes,I should throw an error instead.

---

that would defeat the purpose of using catch? we won't do anything

try {
	release = JSON.parse(releaseJson);
} catch  {}

---

Since this is a JS-based error, I think we should still throw a simple Error instead of a LexActivatorException which has a message along the following lines:

> lexactivator-js encountered an error parsing a Release from the server. Please contact support with your environment details (stack trace, runtime, runtime version, OS, etc.).

Not throwing an error at all can cause the user to not know what is wrong with the application at all and lead to long debugging cycles. I think that is much worse than a prompt error.